### PR TITLE
VideoPlayer + win32: fix video stuttering 

### PR DESCRIPTION
**Pull request by FernetMenta**
1) fix http://trac.kodi.tv/ticket/17100 and forum
http://forum.kodi.tv/showthread.php?tid=299115&pid=2488633#pid2488633

2) allow dr kicking in when video player is paused

### CODE REVIEW

**Comment by FernetMenta** [comment]
Note that calling decode for dxva does not take long, it just queues the commands for the GPU. That means that we don't really block the render tread with this change.

File: xbmc/rendering/dx/RenderSystemDX.cpp [hunk 1]
```diff
@@ -1189,6 +1189,17 @@ void CRenderSystemDX::PresentRenderImpl(bool rendered)
...
+    XbmcThreads::EndTime timer;
+    timer.Set(5);
+    while (!m_decodingTimer.IsTimePast() && !timer.IsTimePast())
+    {
+      m_decodingEvent.wait(lock, 1);
+    }
+  }
+
   FinishCommandList();
   m_pImdContext->Flush();
```
Comment (afedchin) [inline xbmc/rendering/dx/RenderSystemDX.cpp]: you were going to drop this. don't you?

Comment (FernetMenta) [inline xbmc/rendering/dx/RenderSystemDX.cpp]: drop flush? you said that it does not harm so I kept it.

Comment (afedchin) [inline xbmc/rendering/dx/RenderSystemDX.cpp]: ok. keep it as is.

**Review by afedchin — APPROVED** [review]

**Comment by FernetMenta** [comment]
jenkins build this please

**Pull request merged**